### PR TITLE
✨clusterctl: fix RBAC rules and make multi-tenancy possible

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR implements a transformation in the RBAC rules read from the components YAML that ensures that also the ClusterRole and the ClusterRoleBindings are duplicated for each instance of a provider.

By having duplicated RBAC each instance of the provider can have separated lifecycles without introducing any backward/forward compatibility rules.

Nb. technically the providers share CRD, but CRD changes are strictly ruled.

**Which issue(s) this PR fixes** 
Fixes: https://github.com/kubernetes-sigs/cluster-api/issues/1631
Rif #1729

@vincepri 
@ncdc

### CODE REVIEW

File: cmd/clusterctl/pkg/client/repository/components.go [hunk 1]
```diff
@@ -339,11 +341,28 @@ func isResourceNamespaced(kind string) bool {
 	}
 }
 
+const clusterRoleKind = "ClusterRole"
 const clusterRoleBindingKind = "ClusterRoleBinding"
+const roleBindingKind = "RoleBinding"
```
Comment (vincepri) [inline cmd/clusterctl/pkg/client/repository/components.go]: Convert to a `const ( ... )` block?

**Comment by vincepri** [comment]
Looking

File: cmd/clusterctl/pkg/client/repository/components.go [hunk 2]
```diff
@@ -339,11 +341,28 @@ func isResourceNamespaced(kind string) bool {
...
+			newName := fmt.Sprintf("%s-%s", targetNamespace, currentName)
+			o.SetName(newName)
+
+			renamedClusterRoles[currentName] = newName
+		}
+	}
+
+	for i, o := range objs {
 		// if the object has Kind ClusterRoleBinding
 		if o.GetKind() == clusterRoleBindingKind {
```
Comment (vincepri) [inline cmd/clusterctl/pkg/client/repository/components.go]: Maybe convert these ifs to a switch?

**Comment by fabriziopandini** [comment]
@vincepri comment addressed

**Pull request merged**